.github/workflows/benchmark-substrate.yml

```diff
@@ -0,0 +1,200 @@
+name: BENCHMARK-SUBSTRATE
+on:
+  workflow_dispatch:
+    inputs:
+      benchmark-time:
+        description: 'Benchmark time(h/m/s)'
+        default: 30m
+        require: true
+      indexerVersion:
+        description: 'indexer version'
+        default: 'latest'
+        require: true
+      deployment:
+        description: 'Deployment'
+        default: 'QmakYyTWzTxaDU842CczyDWV51Ay9E55k7kbRRXTB3f395'
+        require: false
+      network-endpoint:
+        description: 'Network endpoint'
+        default: 'wss://rpc-polkadot.luckyfriday.io'
+        require: true
+      disableHistorical:
+        description: 'Disable historical'
+        default: 'true'
+        require: false
+      batch-size:
+        description: 'batch-size'
+        default: 30
+        require: false
+      workers:
+        description: 'workers'
+        default: 4
+        require: false
+      others:
+        description: 'Other flags'
+        require: false
+
+jobs:
+  # Label of the container job
+  container-job:
+    # Containers must run in Linux based operating systems
+    runs-on: ubuntu-latest
+    continue-on-error: true
+    # Docker Hub image that `container-job` executes in
+    container: node:18
+
+    # Service containers to run with `container-job`
+    services:
+      # Label used to access the service container
+      postgres:
+        # Docker Hub image
+        image: postgres:12-alpine
+        # Provide the password for postgres
+        env:
+          POSTGRES_PASSWORD: postgres
+        # Set health checks to wait until postgres has started
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+
+    steps:
+      #Check out
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 100
+          token: ${{ secrets.REPO_TOKEN }}
+
+      - name: Install PostgreSQL client
+        run: |
+          apt-get update
+          apt-get install --yes postgresql-client
+
+      # queries database with postgres client
+      - name: Setup historical database
+        run: psql -h postgres -d postgres -U postgres -c 'CREATE EXTENSION IF NOT EXISTS btree_gist;'
+        env:
+          PGPASSWORD: postgres
+
+      - name: Setup Node.js environment
+        uses: actions/setup-node@v2
+        with:
+          node-version: 18
+
+      - name: Install dependencies
```

Comment:
Is it worth moving this to the script so we can have 1 place to update this to other sdks?